test/libsolidity/syntaxTests/constants/constant_with_dependencies_on_constants.sol

```diff
@@ -0,0 +1,35 @@
+uint256 constant MAX = 1;
+
+library L1 {
+    uint256 internal constant INT = 100;
+}
+
+contract C1 {
+    uint256 internal constant CONST = 20 + L2.INT; // forward reference
+    uint256 internal constant LIMIT = MAX * L1.INT;  // same file & external library constant
+    uint256 internal constant NESTED = LIMIT + CONST; // nested & same contract constant
+
+    uint256[L1.INT] internal arr1; // error, backward reference
+    uint256[L2.INT] internal arr2; // error, forward reference
```

Comment:
Can you put the part testing constants in array sizes in a separate test? The dependencies that we were testing before need a positive test that causes no errors. Some errors are fatal so if there's an error, you cannot be 100% sure that there would be no other errors if you removed it.

You are really testing two different things here. They should be in separate files named in such a way that we can easily discern them.

Reply:
Agree, done.

Reply:
I meant something slightly different, i.e.:
1) A passing test that only tests dependencies (i.e. basically the test you had originally).
2) A failing tests that uses constants with dependencies as array sizes.

Reply:
Pushed changes to resolve this comment